eng/pipelines/templates/stages/archetype-c-release.yml

```diff
@@ -18,6 +18,10 @@ stages:
             deploy:
               steps:
                 - checkout: self
+                - pwsh: |
+                    $pkgInfo = Get-Content $(Pipeline.Workspace)/release-info/package-info.json | ConvertFrom-Json
```

Comment:
Can we convert this to:
```
  - task: Powershell@2
    inputs:
      filePath: $(Build.SourcesDirectory)/eng/common/scripts/Verify-ChangeLog.ps1
      arguments: >
        -ChangeLogLocation $(Pipeline.Workspace)/release-info/CHANGELOG.md
        -VersionString ($(Build.SourcesDirectory)/eng/scripts/Get-SdkVersion.ps1)
        -ForRelease $true
      pwsh: true
    displayName: Verify CHANGELOG.md contents
```
Also can you help me understand the relationship between Get-SdkVersion.ps1 and the package-info.json file? Do we still need the package-info.json file?

Reply:
This design takes inspiration from the way we do repo tagging for packages in other languages. When creating the release artifact it gathers up the information that is interesting to the release and provides it in a format that we can parse. Instead of opening up a package file and parsing some contents, the release metadata is left in a json format. There's nothing stopping us from using `Get-SdkVersion.ps1` here instead of the release artifacts. 

The package-info.json file is convenient in that it decouples publishing steps from other steps which retrieve release metadata. It also prevents having to run metadata retrieval steps multiple times. If there was an interface change (or file move) of `Get-SdkVersion.ps1` that change would have to propagate to uses of `Get-SdkVersion.ps1`. Conversely any breaking change to the schema of the artifacts would have pretty far-reaching consequences. 

There is an example of something like this happening here: https://github.com/Azure/azure-sdk-for-c/pull/977#issuecomment-669524468 ... In that example because release scripts were expecting the CHANGELOG.md information to be in a specific place in the artifacts the fact that the CHANGELOG.md file moved was inconsequential as long as it was in the expected place in the artifacts. 

I have no strong feelings about either approach.

Reply:
>The package-info.json file is convenient in that it decouples publishing steps from other steps which retrieve release metadata. It also prevents having to run metadata retrieval steps multiple times. If there was an interface change (or file move) of Get-SdkVersion.ps1 that change would have to propagate to uses of Get-SdkVersion.ps1. Conversely any breaking change to the schema of the artifacts would have pretty far-reaching consequences.

I'm not sure I agree with this as you still have to retrieve/parse this metadata in each step that needs the information. As for interface changes that holds true for everything, including if we move the output location of the package-info.json file. No matter what we use it could potentially move or change. 

From a release prospective I could get behind using the package-info.json metadata file, I'm mostly trying to figure some sort of guideline that would would use to explain to people when to use it over using the other versioning scripts. Would you suggest a guideline that says package-info.json should only be read in release steps? For the other repo's currently I believe this package metadata is only ever used in the release-tagging scripts, the other places like version bumping and changelog validation end up using the versions checked into the repo. 

I will say I do kind of like the idea of having the release processes only work on the artifacts and not depend on the source changes in the repo but we are still going to need to sync the repo for the tools/scripts for releases so we will have both options available. 

If we do end up continuing to using package-info.json, which I'm not opposed to, then we should still use the Powershell task instead of pwsh shortcut to help with diagnosing issues so add the parameter `-VersionString (Get-Content $(Pipeline.Workspace)/release-info/package-info.json | ConvertFrom-Json).version`